### ISSUE TITLE
URL Encode path before attempting to resolve it

### ISSUE THREAD

**Issue by ndw** [ndw/xmlcalabash1#272]
Via email:

```
I tried today to integrate the new 1.1.20-98 version of the Calabash engine.
All the integration tests are failing with the following exception:

java.lang.IllegalArgumentException: Illegal character in opaque part at index 2:
D:\EclipseWorkspaces\WorkspaceSyncRo\eXml\classes
at java.net.URI.create(URI.java:852)
at java.net.URI.resolve(URI.java:1036)
at com.xmlcalabash.core.XProcConfiguration.init(XProcConfiguration.java:223)
at com.xmlcalabash.core.XProcConfiguration.<init>(XProcConfiguration.java:145)
at ro.sync.xml.transformer.xproc.calabash.XProcTransformerImpl$2.run(XProcTransformerImpl.java:208)
at ro.sync.xml.transformer.xproc.calabash.XProcTransformerImpl$2.run(XProcTransformerImpl.java:197)
at java.security.AccessController.doPrivileged(Native Method)
at ro.sync.xml.transformer.xproc.calabash.XProcTransformerImpl.prepareRuntime(XProcTransformerImpl.java:197)
at ro.sync.xml.transformer.xproc.calabash.XProcTransformerImpl.transform(XProcTransformerImpl.java:310)
at ro.sync.xml.transformer.xproc.XProcTransformer$4.run(XProcTransformer.java:562)
at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.URISyntaxException: Illegal character in opaque part at index 2:
D:\EclipseWorkspaces\WorkspaceSyncRo\eXml\classes
at java.net.URI$Parser.fail(URI.java:2848)
at java.net.URI$Parser.checkChars(URI.java:3021)
at java.net.URI$Parser.parse(URI.java:3058)
at java.net.URI.<init>(URI.java:588)
at java.net.URI.create(URI.java:850)
... 10 more

Please take into account to encode the "path" before asking an URI to "resolve" it.

I patched the code as seen bellow and works fine:
--------------------------------------
String classPath = System.getProperty("java.class.path");
String[] pathElements = classPath.split(System.getProperty("path.separator"));
for (String path : pathElements) {
// Start OXYGEN PATCH
try {
path = URLEncoder.encode(path, "UTF-8");
} catch (UnsupportedEncodingException e1) {
// Neglect
}
// End OXYGEN patch

// Make the path absolute wrt the cwd so that it can be opened later regardless of context
String s = cwd.resolve(path).getPath();
```

**Comment by bertfrees** [comment]
Something is not quite right yet in this code to resolve a jar against the cwd on Windows. We're getting this error:

```
java.nio.file.InvalidPathException: Illegal char <:> at index 68: C:\Users\trfre\Documents\Github\pipeline-scripts\epub3-to-daisy202\c:\Program+Files\apache-maven-3.6.0\bin\..\boot\plexus-classworlds-2.5.2.jar
        at java.base/sun.nio.fs.WindowsPathParser.normalize(Unknown Source)
        at java.base/sun.nio.fs.WindowsPathParser.parse(Unknown Source)
        at java.base/sun.nio.fs.WindowsPathParser.parse(Unknown Source)
        at java.base/sun.nio.fs.WindowsPath.parse(Unknown Source)
        at java.base/sun.nio.fs.WindowsFileSystem.getPath(Unknown Source)
        at java.base/java.io.File.toPath(Unknown Source)
        at java.base/java.util.zip.ZipFile$Source.get(Unknown Source)
        at java.base/java.util.zip.ZipFile$CleanableResource.<init>(Unknown Source)
        at java.base/java.util.zip.ZipFile$CleanableResource.get(Unknown Source)
        at java.base/java.util.zip.ZipFile.<init>(Unknown Source)
        at java.base/java.util.zip.ZipFile.<init>(Unknown Source)
        at java.base/java.util.jar.JarFile.<init>(Unknown Source)
        at java.base/java.util.jar.JarFile.<init>(Unknown Source)
        at java.base/java.util.jar.JarFile.<init>(Unknown Source)
        at com.xmlcalabash.core.XProcConfiguration.init(XProcConfiguration.java:240)
        at com.xmlcalabash.core.XProcConfiguration.<init>(XProcConfiguration.java:159)
```

**Comment by bertfrees** [comment]
@ndw I think I have fixed this. See commit https://github.com/daisy/xmlcalabash1/commit/990befdfa53a51289779e0d77e16122c4804f85f.

**Comment by ndw** [comment]
Ok. I've applied that patch. I modified it to use the utf-8 encoding explicitly, as recommended. That avoids a call to a deprecated method.

Please give the current 97 or 98 branches a whirl and let me know if you're satisfied.

**Comment by bertfrees** [comment]
I don't see any differences between your and my commit?

**Comment by ndw** [comment]
I added the encoding to the `.decode()` method call.

**Comment by bertfrees** [comment]
OK I see now, it's in the next commit :)

**Comment by bertfrees** [comment]
OK looks good!

**Comment by ndw** [comment]
Fixed in 1.1.25, I believe.
